docs/how-to-guides/develop-a-flow/develop-standard-flow.md

```diff
@@ -310,4 +310,75 @@ Since `convert_to_dict` has a dictionary output with two keys: `category` and `e
 ![flow_output](../../media/how-to-guides/develop-standard-flow/flow_output.png)
 :::
 
-::::
\ No newline at end of file
+::::
+
+## Referencing external files/folders in a flow
+
+Sometimes, pre-existing code assets are essential for the flow reference. It's common to reference external files or folders that are located outside of the current flow folder. To facilitate this, we allow you to reference these external files or folders using **additional includes** in your flow configuration.
+
+This feature provides an efficient mechanism to list relative file or folder paths that are outside of the flow folder, integrating them seamlessly into your flow.dag.yaml. For example:
+
+```yaml
+additional_includes:
+- ../web-classification/classify_with_llm.jinja2
+- ../web-classification/convert_to_dict.py
+- ../web-classification/fetch_text_content_from_url.py
+- ../web-classification/prepare_examples.py
+- ../web-classification/summarize_text_content.jinja2
+- ../web-classification/summarize_text_content__variant_1.jinja2
+```
+
+You can add this field `additional_includes` into the flow.dag.yaml. The value of this field is a list of the **relative file/folder path** to the flow folder.
+
+Just as with the common definition of the tool node entry, you can define the tool node entry in the flow.dag.yaml using only the file name, eliminating the need to specify the relative path again. For example:
+
+```yaml
+nodes:
+- name: fetch_text_content_from_url
+  type: python
+  source:
+    type: code
+    path: fetch_text_content_from_url.py
+  inputs:
+    url: ${inputs.url}
+- name: summarize_text_content
+  use_variants: true
+- name: prepare_examples
+  type: python
+  source:
+    type: code
+    path: prepare_examples.py
+  inputs: {}
+```
+
+The entry file "fetch_text_content_from_url.py" of the tool node "fetch_text_content_from_url" is located in "../web-classification/fetch_text_content_from_url.py", as specified in the additional_includes field. The same applies to the "summarize_text_content" tool nodes.
+
+> **Note**:
+>
+> 1. If you have two files with the same name located in different folders specified in the `additional_includes` field, and the file name is also specified as the entry of a tool node, the system will reference the **last one** it encounters in the `additional_includes` field.
+> 1. If you have a file in the flow folder with the same name as a file specified in the `additional_includes` field, the system will prioritize the file listed in the `additional_includes` field.
```

Comment:
@lalala123123 is this the designed behavior? I thought we will raise some error?

Reply:
Agreed, I feel like the experience is a bit confusing right now, @lalala123123  could we have some warning?

Reply:
Yes, this is a reference to the behavior of the V2 SDK. In this case, there will be a warning log.
https://github.com/microsoft/promptflow/blob/67adf589d9b3ec54f4721023030efea58d3d49ae/src/promptflow/promptflow/_sdk/_utils.py#L492C17-L495C18